go/engine/identify2_with_uid.go

```diff
@@ -343,10 +343,53 @@ func (e *Identify2WithUID) Run(m libkb.MetaContext) (err error) {
 	return err
 }
 
+func (e *Identify2WithUID) notifyChat(m libkb.MetaContext, inErr error) error {
+	m.Debug("Identify2WithUID.run: notifyChat")
+	if !e.arg.IdentifyBehavior.ShouldRefreshChatView() {
+		return nil
+	}
+	if e.them == nil {
+		return fmt.Errorf("nil user")
+	}
+	nun := e.them.GetNormalizedName()
+	if nun.IsNil() {
+		return fmt.Errorf("no username for user")
+	}
+	// Don't notify if there's a non-identity proof error.
+	if inErr != nil && !libkb.IsIdentifyProofError(inErr) {
+		return nil
+	}
+
+	res, err := e.Result(m)
+	if err != nil {
+		return err
+	}
+
+	update := keybase1.CanonicalTLFNameAndIDWithBreaks{
+		CanonicalName: keybase1.CanonicalTlfName(nun),
+	}
+	if libkb.IsIdentifyProofError(inErr) {
+		update.Breaks = keybase1.TLFBreak{Breaks: []keybase1.TLFIdentifyFailure{
+			keybase1.TLFIdentifyFailure{
+				User:   *e.them.Export(),
+				Breaks: res.TrackBreaks,
+			},
+		}}
+	}
+	m.Debug("Identify2WithUID.run: running HandleChatIdentifyUpdate: %#v", update)
+	m.G().NotifyRouter.HandleChatIdentifyUpdate(m.Ctx(), update)
```

Comment:
Note this doesn't update the chat's CachingIdentifyNotifier's cache. Is that ok? If not, my thoughts are
to make another one of these listeners like HandleUserChanged, but for HandleIdentityFinished. That can update the chat's cache.

Reply:
Should be fine